### PR TITLE
feat: watch the TSTyche config file for changes

### PR DESCRIPTION
**Pull request by mrazauskas**
This PR implements watching for changes of the TSTyche config file.

Left to do:

- [x] handle errors found in changed config file
- [x] warn if no files are selected and wait for changes 
- [x] add tests

### CODE REVIEW

File: source/runner/WatchModeManager.ts [hunk 1]
```diff
@@ -1,8 +1,24 @@
...
+import { CancellationReason, type CancellationToken } from "#token";
+import { Watcher, type WatcherCallback } from "#watch";
+
+class Deferred {
+  promise: Promise<void>;
+  reject!: (reason?: unknown) => void;
+  resolve!: (value: void | PromiseLike<void>) => void;
+
+  constructor() {
+    this.promise = new Promise((resolve, reject) => {
```
[the remark this answers is not in the document]
Comment (mrazauskas) [reply source/runner/WatchModeManager.ts:16]: Thanks. That would be nice indeed. Is TypeScript able to downlevel `withResolvers`? I will check. On MDN it is stated that only Node.js 22 supports it: https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Promise/withResolvers

[the remark this answers is not in the document]
Comment (SerkanSipahi) [reply source/runner/WatchModeManager.ts]: > Is TypeScript able to downlevel withResolvers?

I think so. See https://github.com/microsoft/TypeScript/issues/56483#issuecomment-1947633726 
You probably still need to activate it in compilerOptions.libs. But I'm not quite sure what exactly.

[the remark this answers is not in the document]
Comment (SerkanSipahi) [reply source/runner/WatchModeManager.ts]: What I could find out is that `Promise.withResolvers` is part of `ECMAScript 2024`.

[the remark this answers is not in the document]
Comment (mrazauskas) [reply source/runner/WatchModeManager.ts]: Hm.. I manage to get it typed, but it doesn't seem that there is any downleveling. In the other news, I came back to async watcher, so now a simple `Promise.all()` does the job. That looks much better than my initial solution. So this improved. Thanks for thinking together.

[the remark this answers is not in the document]
Comment (SerkanSipahi) [reply source/runner/WatchModeManager.ts]: For a private project I wrote my own polyfill as long it is not available:

```ts
/**
 * @note:
 * With TypeScript 5.4 we can remove Promise polyfill in order to use the built-in Promise for "withResolvers".
 */
export class Promise {
  static withResolvers<ResponseType>() {
    let resolve: (value: ResponseType | PromiseLike<ResponseType>) => void;
    let reject: (reason?: any) => void;
    const promise = new globalThis.Promise<ResponseType>((_resolve, _reject) => {
      resolve = _resolve;
      reject = _reject;
    });

    // @ts-ignore
    return { promise, resolve, reject };
  }
}
```

Usage:
```ts
const { promise, resolve, reject } = Promise.withResolvers<ResponseType>();
```

Once it is available you could drop the polyfill but it is up to you how you want to deal with it.

**Pull request merged**